lib/ansible/modules/apt_repository.py

```diff
@@ -64,12 +68,20 @@
             - Sets the name of the source list file in sources.list.d.
               Defaults to a file name based on the repository source url.
               The .list extension will be automatically added.
+        type: str
         version_added: '2.1'
     codename:
         description:
             - Override the distribution codename to use for PPA repositories.
               Should usually only be set when working with a PPA on a non-Ubuntu target (e.g. Debian or Mint)
+        type: str
         version_added: '2.3'
+    install_python_apt:
+        description:
+            - Whether to install python apt or not, default is true
+            - If python-apt Python library is not installed this module will not work.
+        type: bool
+        default: true
```

Comment:
what about the comment in arg_spec ?
```
            # This should not be needed, but exists as a failsafe
            install_python_apt=dict(type='bool', default=True),
```
i think this parameter was removed from the doc intentionally

Reply:
It's a good question. This comment has been there like forever (I checked quite some years back in history). For this we probably need core team input, or input from someone who knows the history of this module well.

Reply:
On minimal Debian `python-apt` is not installed for example and right now ansible shows a warning **after** it installs `python-apt` on the server being managed by ansible which would be a problem for someones which are sensitive about packages installed on their server.

I mean there would be a situation where a sysadmin would prefer installing a package herself instead of installing `python-apt` and the package from the playbook.

It should be mentioned in documentation that by default this module installs `python-apt` and people should have the choice whether to disable installing it or not.

Reply:
I know what you mean, and I think it's better to document this properly, but I think this PR is the wrong place for it since this needs further discussion, which would delay the whole PR unnecessarily. It's better to create a new PR for only this change (once this one is merged f.ex., or already now if you don't mind that you might have to resolve conflicts), so that it can be properly discussed.

Reply:
Fine, I will revert the changes to way `devel` branch is now and will create a new PR and link it here.

Reply:
As @felixfontein said it got removed from this PR and it can be further discussed in https://github.com/ansible/ansible/pull/70462.